zproject/prod_settings_template.py

```diff
@@ -120,12 +120,43 @@
     # 'zproject.backends.GoogleAuthBackend',  # Google auth, setup below
     # 'zproject.backends.GitHubAuthBackend',  # GitHub auth, setup below
     # 'zproject.backends.GitLabAuthBackend',  # GitLab auth, setup below
+    # 'zproject.backends.AppleAuthBackend',  # Apple auth, setup below
     # 'zproject.backends.AzureADAuthBackend',  # Microsoft Azure Active Directory auth, setup below
     # 'zproject.backends.SAMLAuthBackend', # SAML, setup below
     # 'zproject.backends.ZulipLDAPAuthBackend',  # LDAP, setup below
     # 'zproject.backends.ZulipRemoteUserBackend',  # Local SSO, setup docs on readthedocs
 )  # type: Tuple[str, ...]
 
+########
+# Apple OAuth.
+#
+# To set up Sign In with Apple, you'll need to do the following:
+#
+# (1) Visit https://developer.apple.com/account/resources/,
+# Follow the instructions in
+# https://help.apple.com/developer-account/?lang=en#/dev77c875b7e
+# to create a Sign In with Apple private key.
+#
+# (2) Follow instructions in
+# https://help.apple.com/developer-account/?lang=en#/dev1c0e25352.
+# to Create a Services ID. In services ID, fill the redirect URI field
+# with a value like `http://zulip.example.com/complete/apple-id`.
+#
+# (3) You'll now have a "Service ID"(eg: com.application.your),a "Key Identifier"
+# and a Private key file. Use the "Service ID" as `SOCIAL_AUTH_APPLE_ID_CLIENT`
+# and "Key Identifier" as `SOCIAL_AUTH_APPLE_ID_KEY` and if you have a "Team ID"
+# that can be set as `SOCIAL_AUTH_APPLE_ID_TEAM` here.
+#
+# (4) Put the Private key file in zulip server in
+# /etc/zulip/apple/zulip-private-key.key. Make sure to set proper permissions
+# with the following commands
+# chown -R zulip:zulip /etc/zulip/apple/
+# chmod 640 /etc/zulip/apple/zulip-private-key.key
+#
+#SOCIAL_AUTH_APPLE_ID_CLIENT = <your Services ID>
+#SOCIAL_AUTH_APPLE_ID_KEY = <your Key ID>
```

Comment:
Do you want a key *identifier* here, or an actual key?

Some of the other code in this PR looks like it wants to interpret this as an actual key. (Or in other words: a secret.) But these instructions say to put the mere key identifier here, which is just a kind of name for a key.

If the instructions are right, the name should agree with it. If not, vice versa :slightly_smiling_face:

Reply:
... Oof, I see, the confusion is there in the [upstream code we're using](https://github.com/python-social-auth/social-core/pull/409/files):
```python
    def generate_client_secret(self):
￼        now = int(time.time())
￼        client_id = self.setting("CLIENT")
￼        team_id = self.setting("TEAM")
￼        key_id = self.setting("KEY")
￼        private_key = self.get_private_key()
```

So be it, then.